Dict key values correctly support null/empty boxes

jvenchy opened the pull request:
## Proposed Changes

_( MemoryViz doesn’t currently support null key values for partial diagrams in a way that can result in empty boxes. I want to adjust this by adding support through blank or whitespace strings. I also wrote three new tests that passed for this feature.)_

...

<details>
<summary>Screenshots of your changes (if applicable)</summary>
original:
![Screenshot 2025-05-14 at 12.12.59 PM.png](attachment:8defa2c7-761b-42fe-ad06-00be5a8712d5:Screenshot_2025-05-14_at_12.12.59_PM.png)
updated:
![Screenshot 2025-05-15 at 1.14.05 AM.png](attachment:2e98007e-427e-4b95-8108-e7b5f755a41f:Screenshot_2025-05-15_at_1.14.05_AM.png)
</details>

![Screenshot 2025-05-14 at 12.41.44 PM.png](attachment:1bd657a8-e199-4daf-9a74-eaaa295aae8a:Screenshot_2025-05-14_at_12.41.44_PM.png)

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |     X    |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |          |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [X] I have performed a self-review of my changes.
    - Check that all changed files included in this pull request are intentional changes.
    - Check that all changes are relevant to the purpose of this pull request, as described above.
- [X] I have added tests for my changes, if applicable.
    - This is **required** for all bug fixes and new features.
- [X] I have updated the project documentation, if applicable.
    - This is **required** for new features. (No new features, not applicable.)
- [X] If this is my first contribution, I have added myself to the list of contributors.
- [X] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

- [] I have verified that the CI tests have passed.
- [x] I have reviewed the test coverage changes reported by Coveralls.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

david-yz-liu commented on memory-viz/src/memory_model.ts:
```diff
@@ -590,7 +590,7 @@ export class MemoryModel {
         for (const k in obj) {
-            let idk = k === null ? "" : `id${k}`;
+            let idk = k.trim() === "" ? "" : `id${k}`;
             let idv = k === null || obj[k] === null ? "" : `id${obj[k]}`;
```
I think for consistency we can remove the `k === null` check here.

jvenchy replied:
Good point, small oversight 😅.

david-yz-liu reviewed (commented):
@jvenchy good work. I left one comment; I also fixed the merge conflict, so you'll need to do a pull from your own branch (`origin dict_values`) to get that commit.

Also make sure to add your name to the list of contributors: this is part of the "Checklist" that you should be filling out before requesting a review.

jvenchy commented:
Hi David, first of all, thanks for all the feedback. I'm getting a better understanding of the pull request workflow for this team so I really appreciate it.

I fixed the code inconsistency and also re-tested my work before committing. I realized my package.json with the contributor list hadn't actually been added to the commit when I had added it, so I fixed that too.
